### PR TITLE
Long Awaited Psychiatrist Rework

### PR DESCRIPTION
**Pull request by ynot01**
# Document the changes in your pull request

The psychiatrist is now a serious role with serious, dangerous equipment. Unfortunately, this is going to be a controversial one.

## Therapy

Psychiatrist can now detect (via examine) and apply therapy to cure traumas by flashing, shining laser pointers or flashlights into peoples' eyes, or just hugging them, because sometimes all you need is a hug. More effective if the patient is susceptible to hypnosis!

## Drugs

Psychiatrists now have a **chemistry lab in their office**. The intent is to place them squarely in the "drug gimmick" role that chemistry sometimes took part in. This also enables them to create their own hallucinogens or otherwise useful chemicals for therapy as mentioned in the wiki documentation. Psychiatrists should not be making medicines very often like a chemist would and should focus instead on their own job. I would consider a psychiatrist making things like libital and distributing them freely without it being related to a patient case or keeping them for personal use to be powergaming.

## Office improvements

Psychiatrists have been equipped with some extra things for therapy, including a couple flashes. Now it makes sense for a psych to be carrying around a (secretly hypno) flash.

All offices have been regulated to carry about the same equipment as box does for consistency.

Their offices have also been **expanded** (good luck everyone else!) (boxstation size unchanged) and divided from the public eye so they can organize themselves in private without patients seeing and having any illusions shattered. Their "Filing Room" (the filing cabinet is the only thing visible from the outside) is fitted with their tools they can prepare gather for effective therapy or drug distribution. Also hidden from the security cameras, sneaky!

I could not fit a chem heater in meta or asteroid so they got a lighter instead.

# Wiki Documentation

Traumas on-gain have their therapy cure chance adjusted 60%-150% to make it sometimes harder or easier to cure traumas.

Magic traumas cannot be cured via therapy

Special traumas have a 15% baseline chance, being meant to be cured before other traumas

Mild traumas have a 10% baseline chance

Severe traumas have a 5% baseline chance

All therapies will have a 1.5x modifier if the patient is "hypnosis vulnerable" (not mindshielded) (hallucinating or sleeping or stupid trauma or unstable mood)

---

Flash therapy can be attempted every 5 minutes

Uses the random cure chance directly, but non-psychiatrists will have a 0.1x modifier

---

Laser (pointer) therapy can be attempted every 8 minutes

About 0.91x effective as flashing, non-psychiatrists will have a 0.1x modifier

---

Shining light into someones eyes (or light therapy) can be attempted every 3 minutes

About 0.83x effective as flashing, non-psychiatrists will have a 0.1x modifier and using a non-penlight flashlight will suffer a 0.8x modifier

---

Hugging therapy can be attempted every 1 minute

About 0.58x as effective as flashing, non-psychiatrists will have a 0.29x modifier and "friendly" huggers will have a 1.25x modifier

# Changelog

:cl:  ynot01, Wejengin2
experimental: Psychiatrist has been reworked. They can now examine to detect traumas and have a wide variety of tools to cure them.
/:cl:

### CODE REVIEW

**Comment by MajManatee** [comment]
I thought this was something else and got REALLY excited.
I guess this is fine. cant say it was something i ever really considered though

**Comment by cowbot92** [comment]
Honestly pretty decent rework. Iffy with powergaming/validhunting but we can see how it works.

**Comment by UltraBigRat** [comment]
better then nothing i suppose, would probably be better if they had more ways to cure traumas or do something when they arent making drugs, other then just blind in three different ways  then hug.

**Comment by Sniblet** [comment]
I feeeeeel like the chem lab is going to be used the wrong way a whole lot. It probably won't that hard to narrow down what chems it doesn't need for psychiatric purposes and remove them.
It could also dispense psychiatric chems directly and have a really small battery.

**Comment by ynot01** [comment]
> I feeeeeel like the chem lab is going to be used the wrong way a whole lot. It probably won't that hard to narrow down what chems it doesn't need for psychiatric purposes and remove them. It could also dispense psychiatric chems directly and have a really small battery.

copy pasted from what i wrote on discord:

> i cant see myself making any compromises regarding the chem setup
> the heater is pretty much qol, not being able to make pills would be silly (are you just going to have your patients sip from a beaker?)
> making a snowflake chem dispenser suited to psych needs would feel incredibly hugboxxy and take away any difficulty from the role
> like wow i need some lsd guess ill press the lsd button
> why even havr the dispenser at that point
> im fully aware of how shit this can turn out, i know full well that people might start carrying heal mixes and going full powermax on psych treating it as chemist with a flash, making nades and other stuff
> its also much more secluded than the chemistry lab so you will be able to make anything you want without any interference or questioning
> its a breeding ground for shitty situations like that
> i can only hope it develops nicely 

TL;DR a special chem dispenser with lsd chems at the press of a button is the most braindead boring hugbox shit imaginable, and im fully aware of the potential consequences of this PR

it helps a bit that psych has a player age requirement now

**Review by patpol4 — APPROVED** [review]
map review done in dms

**Review by patpol4 — APPROVED** [review]
Incredible mapping thank you for your service

**Review by ToasterBiome — APPROVED** [review]
map review map review

i hate the 1 tile maints but they're fine :(

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 1]
```diff
@@ -416,6 +416,8 @@
 				SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "friendly_hug", /datum/mood_event/lamphug, src)
 		for(var/datum/brain_trauma/trauma in M.get_traumas())
 			trauma.on_hug(M, src)
+		for(var/datum/brain_trauma/trauma in get_traumas())
```
Comment (ThatLing) [inline code/modules/mob/living/carbon/carbon_defense.dm:419]: Same code as line above?

File: code/datums/brain_damage/brain_trauma.dm [hunk 2]
```diff
@@ -61,4 +72,69 @@
...
+	if(HAS_TRAIT(hugger.mind, TRAIT_PSYCH))
+		cure_chance *= 3.5
+	if(HAS_TRAIT(hugger, TRAIT_FRIENDLY))
+		cure_chance *= 1.25
+	if(iscarbon(hugged))
+		var/mob/living/carbon/C = hugged
+		if(C.hypnosis_vulnerable())
+			cure_chance *= 1.5
+	if(prob(cure_chance))
+		qdel(src) // Sometimes, all you need is a good hug..
```
Comment (ThatLing) [inline code/datums/brain_damage/brain_trauma.dm:86]: Code is duplicated starting from the ``iscarbon`` check, make a new proc where you pass in the ``cure_chance`` and ``hugger``

**Review by ThatLing — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/carbon_defense.dm (hunk 1 above)
Comment (ynot01) [reply code/modules/mob/living/carbon/carbon_defense.dm:419]: M.get_traumas() vs get_traumas

**Review by ThatLing — APPROVED** [review]

**Comment by wonderinghost** [comment]
By special traumas do you mean deep rooted traumas? sorry if I seem kinda dumb  but my experience with traumas is through the two brain surgeries

**Pull request merged**